### PR TITLE
Refactors events page and updates resampling page

### LINKED ISSUE #10: Replace screenshots in handling event or rewrite section if better

### LINKED ISSUE #19: Update resampling to reflect the resample Params changes
The former resample params has been split up into resampleMinAngle and resampleNPoints so update resampleing.rst appropriately

### PR DESCRIPTION
**Pull request by StephenNneji**
Fixes #19
Fixes #10

### CODE REVIEW

File: source/advanced/resampling.rst [hunk 1]
```diff
@@ -106,12 +108,10 @@ The resampling itself is controlled by the 'resamPars' field in the controls blo
...
-These have the following effects:
+* **resampleMinAngle**: The algorithm refines near the points which form, together with their left and right neighbours, a triangle with central angle smaller than the given value. So, a larger value of *resampleMinAngle* results in a finer resampling. 
+  *resampleMinAngle* is defined as a multiple of pi (i.e. resampleMinAngle = 0.9 refines where the adjacent points form an angle greater than 0.9 * pi)
 
-* **minAngle**: The algorithm refines near the points which form, together with their left and right neighbours, a triangle with central angle smaller than the given value.  So, a larger value of minAngle results in a finer resampling. minAngle is defined as a multiple of pi (i.e. minAngle = 0.9 refines where the adjacent points form an angle greater than 0.9 * pi)
-
-* **nPoints**: The initial number of domain points (layers). The algorithm will start from this number of layers, and refine until all the *minAngle* criteria are satisfied.
+* **resampleNPoints**: The initial number of domain points (layers). The algorithm will start from this number of layers, and refine until all the *resampleMinAngle* criteria are satisfied.
```
Comment (alexhroom) [inline source/advanced/resampling.rst:114]:
```suggestion
* ``resampleNPoints``: The initial number of domain points (layers) sampled by the algorithm at the start.
```
I think the point on the `resampleMinAngle` criteria should be in the `resampleMinAngle` explanation, as above

File: source/advanced/resampling.rst [hunk 2]
```diff
@@ -106,12 +108,10 @@ The resampling itself is controlled by the 'resamPars' field in the controls blo
...
 
 .. note::
-        Generally, *minAngle* has a larger effect on the eventual resampling than *nPoints*.
+        Generally, **resampleMinAngle** has a larger effect on the eventual resampling than **resampleNPoints**.
```
Comment (alexhroom) [inline source/advanced/resampling.rst:117]:
```suggestion
        Generally, ``resampleMinAngle`` has a larger effect on the eventual resampling than ``resampleNPoints``.
```

**Review by alexhroom — CHANGES_REQUESTED** [review]
looks mostly good! just a few improvements to sentence flow/clarity and maybe some improvements to how resampling is explained while we're here

File: source/advanced/resampling.rst [hunk 3]
```diff
@@ -37,14 +37,16 @@ distance between neighbouring points. So, for an SLD profile of a floating bilay
...
 .. image:: ../images/advanced/layersPic.png
      :width: 700
      :alt: sld and dots...
 
-**Using Resampling in RAT**
-Using resampling on a contrast in RAT is very simple. For any contrast that you want to reample, then simply set the 'resample' flag for that contrast to 'true'
+***********************
+Using Resampling in RAT
+***********************
+Using resampling on a contrast in RAT is very simple. For any contrast that you want to resample, then simply set the 'resample' flag for that contrast to 'true'
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply source/advanced/resampling.rst:49]: Should we use the back quotes on  \`\`true`` considering Python expects  True instead?

[the remark this answers is not in the document]
Comment (alexhroom) [reply source/advanced/resampling.rst:49]: yeah, i was thinking about that when i made the suggestion - although if python expects capital-T True then is "set the flag to 'true'" correct either? I guess it's less specific?

[the remark this answers is not in the document]
Comment (StephenNneji) [reply source/advanced/resampling.rst:49]: Maybe we go with "simply set the ``resample`` flag as shown below"

[the remark this answers is not in the document]
Comment (alexhroom) [reply source/advanced/resampling.rst:49]: sounds good to me

File: source/advanced/resampling.rst [hunk 4]
```diff
@@ -106,12 +108,10 @@ The resampling itself is controlled by the 'resamPars' field in the controls blo
             controls = RAT.Controls()
             print(controls)
 
-The resampPars field has two parameters in it, corresponding to the minimum angle and minimum points (resamPars = [minAngle minPoints]) which to use to resample the profiles.
-These have the following effects:
+* **resampleMinAngle**: The algorithm refines near the points which form, together with their left and right neighbours, a triangle with central angle smaller than the given value. So, a larger value of *resampleMinAngle* results in a finer resampling.
```
[the remark this answers is not in the document]
Comment (alexhroom) [reply source/advanced/resampling.rst]: Okay, how about:
"For each data point, the algorithm compares the two lines between that data point and its neighbouring points on either side. If the internal angle between those lines is smaller than `resampleMinAngle`, then the algorithm will refine over that point. 

In practice, this means that resampling happens for points which are significantly larger or smaller than their neighbours, and `resampleMinAngle` controls the sensitivity of this.

The algorithm terminates once there are no three adjacent points which form an angle smaller than `resampleMinAngle`."

maybe it'd make more sense if we added a picture?

[the remark this answers is not in the document]
Comment (StephenNneji) [reply source/advanced/resampling.rst]: I think this is better and I agree a picture would be helpful. I am still not sure about saying "points which are significantly larger or smaller than their neighbours" as far as I know points do not have size so they can't be larger or smaller. Are you still referring to the angle when you say larger or smaller?

[the remark this answers is not in the document]
Comment (alexhroom) [reply source/advanced/resampling.rst]: I meant 'larger or smaller' as in the function value at that point is larger or smaller. Maybe 'higher or lower' or just 'different'?
graphcurve
Hopefully this picture explains it a bit better, with a figure caption that explains that if `resampleMinAngle` is set to 0.9, the algorithm would resample around x_1 and x_3 but not x_2.

I'm happy to add the image and explanation if you're happy with all that?

[the remark this answers is not in the document]
Comment (alexhroom) [reply source/advanced/resampling.rst]: done - I also noticed that the page said the algorithm resamples over angles *larger* than `resampleMinAngle`: this should be *smaller* [(see the docstring for the algorithm)](https://github.com/RascalSoftware/RAT/blob/master/targetFunctions/common/resampleLayers/adaptive.m), so I fixed all instances of that

**Review by alexhroom — APPROVED** [review]

**Pull request merged**